### DISCUSSION TITLE
[Feature Request] Nemo: Drag & Drop Script Execution

### DISCUSSION THREAD

**Discussion by DarkOwl2** [linuxmint/discussions#632] [Ideas]
In Windows, I can write a .bat and run that on a command line with command line parameters which are accessed internally as %1, %2 etc.

OK, I can do that in Linux too, with a bash script.

In Windows, if I drag a file onto the .bat in the file manager (I happen to use DOpus), it executes the .bat with %1 set to the full name and file path of the dropped file. Drop multiple files, and they are %2, %3 etc.

I would like to do that in Linux Mint Cinnamon, ie using Nemo.  It seems not to be currently possible (I'm using Nemo 5.8.5).  This is yet another thing impeding my migration away from Windows.

This doesn't seem like rocket science, it ought to be straight-forward to provide similar functionality in a file manager (ie Nemo): if drag&drop target is an executable shell script, open a terminal window and run the shell script with the drag&drop payload as command parameters, then close the terminal window.

Thread on Linux Mint forum: https://forums.linuxmint.com/viewtopic.php?t=431751

**Comment by Secret-chest** [comment]
Should work for any executable including bash, binaries and python.

**Reply by DarkOwl2** [reply]
> Should work for any executable including bash, binaries and python.

That's good to know, but please give details exactly how.  I have tried the following, without success:

1. Open folder in Nemo (5.8.5)
2. Folder contains sample file and a bash script marked as executable
3. Click&drag sample file onto script file, nothing happens
4. Open terminal window and type "./script sample.file" and the script runs as expected.

**Reply by Secret-chest** [reply]
> > Should work for any executable including bash, binaries and python.
> 
> That's good to know, but please give details exactly how. I have tried the following, without success:
> 
> 1. Open folder in Nemo (5.8.5)
> 2. Folder contains sample file and a bash script marked as executable
> 3. Click&drag sample file onto script file, nothing happens
> 4. Open terminal window and type "./script sample.file" and the script runs as expected.

It doesn't work now, but if it's implemented it should be like this.

**Comment by DarkOwl2** [comment]
Oh I see.  I thought you were saying it should be working.  Sorry.
